File: app/src/main/java/star/iota/acgrip/ui/item/ItemContract.java
package star.iota.acgrip.ui.item;

import java.util.List;

interface ItemContract {

    interface View {

        void success(List<ItemBean> items);

        void error();

        void noData();
    }

    interface Presenter {
        void request(int type, int page);

        void request(String url, int page);

        void search(String keywords, int page);

        void cancel();
    }
}

File: app/src/main/java/star/iota/acgrip/ui/item/ItemContract.kt
package star.iota.acgrip.ui.item

import star.iota.jptv.base.BasePresenter
import star.iota.jptv.base.BaseView

internal interface ItemContract {

    interface View : BaseView {
        fun success(items: List<ItemBean>)
    }

    interface Presenter : BasePresenter {
        fun request(type: Int, page: Int)

        fun request(url: String, page: Int)

        fun search(keywords: String, page: Int)
    }
}
